### DISCUSSION TITLE
Query API with 'from=<a timestamp>'

### DISCUSSION THREAD

**Discussion by Asiya-Yunusa** [datalust/seq-tickets#2025] [Help]
I will like to know if there is a way to query the HTTP  API with from=<a timestamp>. When ever I query it only returns the current logs.

https://<base-url>/api/events?from=2023-11-21T20:26:00Z

**Comment by nblumhardt** [comment]
Hi @Asiya-Yunusa, thanks for your message.

If you GET `/api/events/resources`, the parameters on each endpoint are described:

```json
{
  "Links": {
    "Self": "api/events/resources",
    "Items": "api/events{?signal,filter,count,startAtId,afterId,fromDateUtc,toDateUtc,shortCircuitAfter,render,clef,permalinkId,background}",
    "InSignal": "api/events/signal{?signal,filter,count,startAtId,afterId,fromDateUtc,toDateUtc,shortCircuitAfter,render,clef,permalinkId,defer,background}",
    "Item": "api/events/{id}{?download,render,clef,permalinkId}",
    "DeleteInSignal": "api/events/signal{?signal,filter,fromDateUtc,toDateUtc}",
    "Tabulate": "api/events/tabulate{?signal,filter,fromDateUtc,toDateUtc,defer,background}",
    "Stream": "wss://logs.datalust.co/api/events/stream{?filter,signal}",
    "Raw": "api/events/raw{?clef}"
  }
}
```

For historical reasons, the parameter you're looking for is (unfortunately 😅) called `fromDateUtc`.

An easier way to interact with the API is to provide a filter, or use the `/api/data?q=` endpoint, where date ranges can be expressed in terms like:

```
@Timestamp >= now() - 1d
```

or

```
@Timestamp >= DateTime('2023-11-21T20:26:00Z')
```

Hope this helps!

**Reply by Asiya-Yunusa** [reply]
Thank you for your response!

I am getting an error when I call with 'fromDateUtc' and 'toDateUtc': {"Error":"An unhandled error occurred while serving the request...

But when I run with only fromDateUtc it just returns the present events.

curl -v -X GET --header 'X-Seq-ApiKey: <api-key>' 'https://<base-url>/api/events?fromDateUtc=2023-11-21T20:26:00Z%toDateUtc=2023-11-21T21:26:00Z'

**Reply by liammclennan** [reply]
The easiest way to do this is to build your query in the Seq UI, and copy the API call out of the browser developer tools. 

To diagnose the error you are seeing you can have a look in the Seq logs (/data/Logs). It looks like the format of the path is incorrect. It should be something like:

```
https://yourseqserver.com/api/events?fromDateUtc=2023-11-22T22:45:00.000Z&toDateUtc=2023-11-23T22:45:00.000Z
```
